cupy/cuda/compiler.py

```diff
@@ -160,7 +160,12 @@ def _get_arch_for_options_for_nvrtc(arch=None):
     # generate cubin (SASS) instead of PTX. See #5097 for details.
     if arch is None:
         arch = _get_arch()
-    if _cuda_hip_version >= 11010 and arch < _get_max_compute_capability():
+    use_ptx = _get_bool_env_variable(
```

Comment:
Could you move this code to module-level to avoid getting env var everytime?

Reply:
FYI, there is a lot of code who calls this every time, so probably we should fix all of these in a separate PR.